TheJumpCloud/jumpcloud-ADMU
# V2.7.0

## Issues
* [CUT-4060](https://jumpcloud.atlassian.net/browse/CUT-4060) - CUT-4060 Release 2.7.0 Improved UI/User experience for ADMU Epic

## What does this solve?

- Replace log terminal with progress form UI that consists of: progress bar, status, migration details, optional log view, rerun app and view log options
- Updated JC logo
- Error mapping that points admin to ADMU troubleshooting page
- TECH DEBT: should not be able to leave domain if not domain bound
- TECH DEBT: fixed issue with unjoining hybrid domain systems
- TECH DEBT: prevent migrating same local hostname and new user
- Added logs for processes by owner
EPIC:
https://jumpcloud.atlassian.net/browse/CUT-3869
Log Process:
https://jumpcloud.atlassian.net/browse/CUT-3852

## Is there anything particularly tricky?

## How should this be tested?
1. Create .exe `.\Build.ps1`
2. Open app and validate updated JC logo in the upper left side of the Form
3. Migrate for failure and validate error messages

- Lock the registry by clicking on the registry profile with _admu in the end in the middle of migration
- Rerun a successful migration and select the same System Account that you selected from last migration (usually have a local path with .`WORKGROUP`)

5. Rerun the app using the new `Rerun` button after success/fail and validate that the app is running as if it was first opened
6. Click on the new `View Log` button after a migration, ensure that it opens the log file
7. Validate that you can't migrate with the same system hostname and Local Account Username: 
<img width="955" alt="image" src="https://github.com/user-attachments/assets/e303316f-a7c9-4d2a-ba34-47c759de8bce">
8. Run ADMU on a hybrid domain joined system and select `Leave Domain`, migrate and validate that the system is unjoined from AD
9. After a successful migration, copy the newly built UWP to C:\Windows\. Login to the migrated user and validate the updated logo is shown and the status text shows: "Completing Account Migration x%"
<img width="1503" alt="image" src="https://github.com/user-attachments/assets/001c1bf5-ef1a-48d0-8c2c-6e737fa8e4e2">


10. Run module: Start-Migration twice, with -AdminDebug $true and $false. Validate that the verbose messaging is being hidden on $false and shown on $true


[CUT-3820]: https://jumpcloud.atlassian.net/browse/CUT-3820?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CUT-4060]: https://jumpcloud.atlassian.net/browse/CUT-4060?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Screenshots
<img width="994" alt="image" src="https://github.com/user-attachments/assets/6d7f12e0-f1cb-41a3-b09b-79591f41a57b">
<img width="990" alt="image" src="https://github.com/user-attachments/assets/24ee8416-ca37-48e6-969e-77f42abf6a59">
<img width="994" alt="image" src="https://github.com/user-attachments/assets/4ac2981c-7b8c-46d0-a53b-f7a49db3de00">
<img width="692" alt="image" src="https://github.com/user-attachments/assets/f6c8f7b2-2550-4996-8668-aff40f19e04b">
<img width="694" alt="image" src="https://github.com/user-attachments/assets/2408c84f-8c8c-43d2-8391-6082491266b5">
<img width="1503" alt="image" src="https://github.com/user-attachments/assets/2d9e6648-657f-46f6-a808-6ce413c884a0">
